usr.sbin/rtadvd/config.c

```diff
@@ -912,6 +912,58 @@ getconfig(struct ifinfo *ifi)
 		}
 		free(dns);
 	}
+
+	/*
+	 * handle pref64
+	 */
+	rai->rai_pref64.p64_enabled = false;
+
+	if ((addr = (char *)agetstr("pref64", &bp))) {
+		if (inet_pton(AF_INET6, addr, &rai->rai_pref64.p64_prefix) != 1) {
```

Comment:
I'm no expert with C, but in general I feel it reads better to start with the success case (`== 1`) in a conditional like this. If my opinion goes against `style(9)` or any other convention, please ignore.

Reply:
i disagree, but i understand where you're coming from.

the problem is that this function is horrible to begin with, with a convoluted control controlled by multiple different 'goto' labels, and is quite difficult to follow.  with that in mind, i think erroring early here makes the new code easier to read than putting the success case first.

the properly fix here is to rewrite the entire code in a more modern style, but that's out of scope for this feature.

Reply:
I get that, thanks for explaining. Probably best to keep it like this, until it can be entirely rewritten.